server/tests/integration_test.rs

```diff
@@ -0,0 +1,998 @@
+    while !line.contains("Starting Apache Arrow Flight") {
+        output
+            .read_line(&mut line)
+            .expect("Could not read line of process.");
+    }
+
+    return process;
+}
+
+/// Create a new Arrow Flight client to access the endpoints in the ModelarDB server.
+fn create_flight_service_client(
+    rt: &Runtime,
+    host: &str,
+    port: u16,
+) -> Result<FlightServiceClient<Channel>> {
+    let address = format!("grpc://{}:{}", host, port);
+
+    rt.block_on(async {
+        let fsc = FlightServiceClient::connect(address).await?;
+        Ok(fsc)
+    })
+}
+
+/// Create a model table in the ModelarDB server with Arrow Flight using the `do_action()` SQL Parser.
+fn create_model_table(
+    rt: &Runtime,
+    mut client: &mut FlightServiceClient<Channel>,
+    table_name: String,
+) -> std::result::Result<(), ()> {
+    let cmd = format!(
+        "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD, tag TAG)",
+        table_name
+    );
+    let action = Action {
+        r#type: "CommandStatementUpdate".to_string(),
+        body: cmd.into_bytes(),
+    };
+
+    rt.block_on(async {
+        let mut _result = client.do_action(Request::new(action)).await;
+
+        Ok(())
+    })
+}
+
+/// Create a normal table without tags in the ModelarDB server with Arrow Flight using the `do_action()` SQL Parser.
+fn create_model_table_without_tags(
+    rt: &Runtime,
+    mut client: &mut FlightServiceClient<Channel>,
+    table_name: String,
+) -> std::result::Result<(), ()> {
+    let cmd = format!(
+        "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD)",
+        table_name
+    );
+    let action = Action {
+        r#type: "CommandStatementUpdate".to_string(),
+        body: cmd.into_bytes(),
+    };
+
+    rt.block_on(async {
+        let mut _result = client.do_action(Request::new(action)).await;
+
+        Ok(())
+    })
+}
+
+/// Create a normal table in the ModelarDB server with Arrow Flight using the `do_action()` SQL Parser.
+fn create_table(
+    rt: &Runtime,
+    mut client: &mut FlightServiceClient<Channel>,
+    table_name: String,
+) -> std::result::Result<(), ()> {
+    let cmd = format!(
+        "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+        table_name
+    );
+    let action = Action {
+        r#type: "CommandStatementUpdate".to_string(),
+        body: cmd.into_bytes(),
+    };
+
+    rt.block_on(async {
+        let mut _result = client.do_action(Request::new(action)).await;
+
+        Ok(())
+    })
+}
+
+/// Generate a [`RecordBatch`] with the current timestamp, a random value and an optional tag.
+fn generate_random_message(tag: Option<String>) -> (RecordBatch) {
+    let timestamp = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("Can't generate the time.")
+        .as_micros() as i64;
+
+    let value = rand::thread_rng().gen_range(0..100) as f32;
+
+    if tag.clone().unwrap_or("".to_string()) == "" {
```

Comment:
Instead of almost duplicating the code to create the schema and record batch, maybe bring the check for whether a tag should be included down and just use to it decide whether or not to add an extra item to the two vecs.

Reply:
I reduced it a bit. Was this what you had in mind or do you have other suggestions?

Reply:
It seems easier to just create the `fields` vec and `columns` vec initially with no tag. Then if there is a tag, add the tag field and tag column to the corresponding vecs. Then finally a single `Schema::new` call and a single `RecordBatch::try_new()` can be made.

Reply:
Kind of fixed. I had trouble creating a Vector of different types of Arc outside the `RecordBatch::try_new` function. Reduced as much as I could. @skejserjensen, is there any way to create a Vector of different types of Arc?

Reply:
As far as I am aware that is not possible as `T`, as far as I know, must be the same for `Vec<Arch<T>>`.  However, there are multiple ways to make `T` the same for all elements in `Vec`. For example, since you are already putting the elements behind a pointer you can store them as [`Trait Objects`](https://doc.rust-lang.org/book/ch17-02-trait-objects.html) if share a common trait without adding additional allocation. If that is not possible you could also create an [`Enum` ](https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html) where each kind stores a different value. Finally it may be possible to use an [`Union`](https://doc.rust-lang.org/reference/items/unions.html), although as far as I am aware Rust's `Union`s mainly exists for C-interop and will require `unsafe` code to use.

Reply:
Thanks for that, noted. For the sake of this purpose, I think the current solution works fine. I will remember to look into this if future improvements have to be made.